### ISSUE TITLE
Custom error handling in std/http/file_server.ts

### ISSUE THREAD

**Issue by wperron** [denoland/deno#8249]
What I have in mind is something along the lines of CloudFront's "Error Pages" settings; Essentially allows for redirecting the request to a certain file when a specific error code is returned and optionally to override the status code. This is typically used to redirect 404s to a nicer 404.html page, or to enable SPAs by redirecting 404s to index.html with a 200.

**Comment by kitsonk** [comment]
`std/http/file_server.ts` is indented to be minimalistic.  A rich set of features are better addressed by a community library.

**Comment by wperron** [comment]
@kitsonk thanks for the reply, I completely agree with you, and is one of the things that I like about the std lib. I think I just poorly expressed my idea here; I think we can make the file_server less of a black box and turn it into something that is more open and extendable, without necessarily adding functionality straight into the existing package.

If we simply took the time to export most of the functions in `std/http/file_server.ts` and added exported interface, we could enable the existing code to be more easily used in community modules _and_ it would keep the std lib simple. For instance, the `html` and `serverLog` (among other) could be exported. the `dirViewerTemplate` and `serveFallback` could be exposed as interfaces that can be implemented in userland as well.

TL;DR: I think we can make the file_server easier to extend while also keeping it simple. wdyt?

**Comment by ry** [comment]
The intention with file_server is that it could be extended for use in other programs - much like https://golang.org/pkg/net/http/#ServeFile
